### PR TITLE
Adds onChangePasswordEmailRequest method

### PR DESCRIPTION
**Pull request by UmamaheshMaxwell**
Adds onChangePasswordEmailRequest for change password

### CODE REVIEW

File: src/authentication.ts [hunk 1]
```diff
@@ -163,6 +163,12 @@ export interface IAuthenticationAuthorityOptions {
   onForgotPasswordEmailRequest: (user: any, token: string, req?: Request, res?: Response) => Promise<any>;
 
   /**
+   * Called when the user requests a "change password" email.
```
Comment (etsuo) [inline src/authentication.ts]: Should be: Called when the user changes his or her password, allowing the integrator to send an email to the user notifying them of the password change.

File: src/authentication.ts [hunk 2]
```diff
@@ -163,6 +163,12 @@ export interface IAuthenticationAuthorityOptions {
...
+   * @param user
+   */
+  onChangePasswordEmailRequest: (user: any, req?: Request, res?: Response) => Promise<any>;
```
Comment (etsuo) [inline src/authentication.ts]: This should be optional (`onChangePasswordEmailRequest?:`) - unlike `onUserCreated`, or `onResendEmailConfirmation`, `onChangePasswordEmailRequest` is an optional notification. It's up to the implementor to decide if he/she wants to send an email (or whatever).

File: src/authentication.ts [hunk 3]
```diff
@@ -390,6 +395,7 @@ export function addAuthenticationAuthority(sapi: SakuraApi, options: IAuthentica
             [fields.passwordStrengthDb]: this.getPasswordStrength(newPassword, user)
           });
         })
+        .then(() => options.onChangePasswordEmailRequest(user, req, res))
```
Comment (etsuo) [inline src/authentication.ts]: This should not fire if it's not defined... that'll generate an error (see comments above about needing to make this optional. Something like

`.then(() => (options.onChangePasswordEmailRequest) ? options.onChangePasswordEmailRequest(user, req, res) : Promise.resolve())`

**Comment by UmamaheshMaxwell** [comment]
Hi @etsuo ,
    Made the code review changes and push the changes, This PR is ready now

**Pull request merged**